### DISCUSSION TITLE
PINN Operation -- extra boot and root partitions being automounted

### DISCUSSION THREAD

**Discussion by dshuman52** [procount/pinn#856] [General]
The following was done on a Ras PI 3 using the current 3.9.3 version of PINN.

I created a PINN boot sd with the three RasPI OSs  Bookworm/64, Bookworm32 and Bullseye32.  Each OS when booted automounts the unneeded boot and root  partitions of the other OSs.  Since my grandson will be using this configuration and to prevent his contamination/interference between OSs I added /etc/fstab entries to each OS to block mounting of the others.  (there are two sets of the following for each extra OS boot & root partition in my setup.)

```
/dev/mmcblk0p6   none   vfat  ro,noauto,noload  0  0    ### block automount of unused boot partition
/dev/mmcbkk0p7   none   ext4  ro,noauto,noload  0  0    ### block automount of unused root partition
```

NOTE: The option to block automount in pcmanfm (the default linux file manager) leaves the items visible in the File Manager GUI presentation making them easily mounted and therefore corruptible.  So I rejected that alternative, in my case.

I am not expert so take this with a grain of salt.; I believe the following:

The ro option is protection if one manages to actually mount the drive which might be possible.  The noauto option achieves most of the intended goal of "hiding" at least partially the partitions associated with the other OS's.  The noload option, if I understand correctly, prevents one PI OS from attempting to cleanup journal entries left over after an ill timed interruption of one of the other RasPI OSs.  Therefore; I expect each RasPI OS will be cleaning up it's own pending journal entries.  Also, lsblk can still see all of the partitions.

I considered that the configuration option 'disablesdimages' might help here.  However; after testing it I now believe the disablesdimages option may be intended to prevent (manual/custom) loading of any OSs into/by PINN that the user may have placed onto the SD card.  If this is true the documentation could probably be improved for disablesdimages and disableusbimages.  Also after using the GUI to test the disablesdimages; it appears these are also 'select=' options.

**Comment by procount** [comment]
> Each OS when booted automounts the unneeded boot and root partitions of the other OSs.

This is a function of the Raspios file manager and the types of drives you use. If you are using a USB drive or SD card, then they are classed as removable, but an SSD or HDD are not. It's not due to PINN, just a consequence of having multiple partitions. THe same would happen if you plugged in additional removable drives.

> NOTE: The option to block automount in pcmanfm (the default linux file manager) leaves the items visible in the File Manager GUI presentation making them easily mounted and therefore corruptible. So I rejected that alternative, in my case.

There is also the option in Edit - PReferences - Layout - Show in Places - Volumes and Mounts as well as the options under Volume Management, which might help.

> I considered that the configuration option 'disablesdimages' might help here.

No they are not related to this. They concern which sources PINN will use to find OSes to INSTALL. By disabling some sources, they can expose OSes that may be older from different sources, or speed up initialization by e.g. not waiting for network sources. Other users wanted options to automatically select and exclude OSes and even auto-install or auto-update OSes with PINN. I don't think they are appropriate for your use cases.

_EDIT_
>If this is true the documentation could probably be improved for disablesdimages and disableusbimages

The headline desciptions of these options are probably a bit terse, but I think they are better described in
https://github.com/procount/pinn/blob/master/README_PINN.md#auto-installing-oses-based-on-their-source

**Comment by dshuman52** [comment]
My reason for adding this to the discussion was to indicate the creator of the multi-boot device had the power themselves to fix this problem on their own (assuming my changes are as effective I as believe they are after further testing).

    On Saturday, December 14, 2024 at 06:31:57 PM EST, procount ***@***.***> wrote:  
 
 



Each OS when booted automounts the unneeded boot and root partitions of the other OSs.


This is a function of the Raspios file manager and the types of drives you use. If you are using a USB drive or SD card, then they are classed as removable, but an SSD or HDD are not. It's not due to PINN, just a consequence of having multiple partitions. THe same would happen if you plugged in additional removable drives.


NOTE: The option to block automount in pcmanfm (the default linux file manager) leaves the items visible in the File Manager GUI presentation making them easily mounted and therefore corruptible. So I rejected that alternative, in my case.


There is also the option in Edit - PReferences - Layout - Show in Places - Volumes and Mounts as well as the options under Volume Management, which might help.


I considered that the configuration option 'disablesdimages' might help here.


No they are not related to this. They concern which sources PINN will use to find OSes to INSTALL. By disabling some sources, they can expose OSes that may be older from different sources, or speed up initialization by e.g. not waiting for network sources. Other users wanted options to automatically select and exclude OSes and even auto-install or auto-update PINN. I don't think they are appropriate for your use cases.
